File: Domain/ChatRoomRole.cs
namespace Domain;

public class ChatRoomRole
{
    public required string Id { get; set; } = Guid.NewGuid().ToString();
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required string Color { get; set; } = "#ffffff"; // Default color white
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool IsDefault { get; set; } = false;

    // Navigation properties
    public required string ChatRoomId { get; set; }
    public ChatRoom ChatRoom { get; set; } = null!;
    public ICollection<ChatRoomMemberRole> MemberRoles { get; set; } = [];
    public ICollection<ChatRoomRolePermission> RolePermissions { get; set; } = [];
}

File: client/src/lib/mappings/chatRoomRole.ts
import type { ChatRoomRole } from "../types";

export function mapChatRoomRoleFromServer(serverRole: any): ChatRoomRole {
  return {
    id: serverRole.id,
    name: serverRole.name,
    description: serverRole.description,
    color: serverRole.color,
    createdAt: serverRole.createdAt,
    isDefault: serverRole.isDefault,
    chatRoomId: serverRole.chatRoomId,
    permissions: (serverRole.permissions || []).map((serverRolePermission: any) => ({
      roleId: serverRolePermission.roleId,
      isAllowed: serverRolePermission.isAllowed,
      id: serverRolePermission.permission.id,
      name: serverRolePermission.permission.name,
      description: serverRolePermission.permission.description
    })),
  };
}